routers/api/v2/router.go

```diff
@@ -0,0 +1,131 @@
+package v2
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/pkg/errors"
+	v2 "github.com/unicsmcr/hs_auth/authorization/v2"
+	"github.com/unicsmcr/hs_auth/entities"
+	"github.com/unicsmcr/hs_auth/routers/api/models"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.uber.org/zap"
+	"net/http"
+	"strings"
+	"time"
+)
+
+type APIV2Router interface {
+	models.Router
+	Login(ctx *gin.Context)
+	GetUsers(ctx *gin.Context)
+	GetAuthorizedResources(ctx *gin.Context)
+}
+
+type apiV2Router struct {
+	models.BaseRouter
+	logger *zap.Logger
+	authorizer v2.Authorizer
+}
+
+
+func NewAPIV2Router(logger *zap.Logger, authorizer v2.Authorizer) APIV2Router {
+	return &apiV2Router{
+		logger: logger,
+		authorizer: authorizer,
+	}
+}
+
+func (r *apiV2Router) RegisterRoutes(routerGroup *gin.RouterGroup) {
+	routerGroup.GET("/", r.Heartbeat)
+
+	usersGroup := routerGroup.Group("/users")
+	usersGroup.GET("/", r.buildAuthChecker("hs:hs_auth:api:v2:getUsers"), r.GetUsers)
```

Comment:
What do you think of having a factory function that would construct these URIs instead of having to manually hardcode them everytime? Don't have to do it in this PR, just to start a discussion really :)

Reply:
Yeah that was my plan for the future, it's just outside of the scope for this PR.

Ultimately I'd want this line to look like this:
`usersGroup.GET("/", r.withAuthMiddleware(r.GetUsers))`
The resource name would be constructed from the path to the router (specified in the router's constructor), the name of the passed function (there's a way to do this with reflection but it might have a significant impact on performance, will need to do some benchmarks) and the path/query/form parameters in the request.

Reply:
I like that idea. As you mentioned, reflection during runtime can have an impact on performance. Maybe it's possible to perform the reflection at startup which creates some sort of mapping from the route to URI to prevent doing it during runtime?

Reply:
Yeah that's a good idea.

Now that I think about it, it would actually work like that already. This `withAuthMiddleware` call would be located within `RegisterRoutes` method which is called to initialize a specific router at startup. Since we're just constructing the handlers here instead of handling the request itself, we'd only be using reflection once per API operation at startup.